code/android-core-library/src/phone/java/com/adobe/marketing/mobile/services/caching/CacheFileManager.java

```diff
@@ -0,0 +1,290 @@
+                     @NonNull final String rootCacheDirName) {
+        this.rootCacheDirName = rootCacheDirName;
+        this.deviceInfoService = deviceInfoService;
+    }
+
+    /**
+     * Creates a directory with name {@code cacheName} inside the {@code cacheRoot}
+     * folder of {@code deviceInfoService.getApplicationCacheDir()}
+     *
+     * @param cacheName the name of the folder to act as a cache bucket.
+     * @return a {@code File} handle for the newly created/existing bucket;
+     * null if the directory creation fails.
+     */
+    File createCache(final String cacheName) {
+        if (StringUtils.isNullOrEmpty(cacheName)) return null;
+
+        final File appCacheDir = deviceInfoService.getApplicationCacheDir();
+        if (!FileUtils.isWritableDirectory(appCacheDir)) {
+            Log.debug(ServiceConstants.LOG_TAG, TAG, "App cache directory is not writable.");
+            return null;
+        }
+
+        final File cacheBucket = new File(appCacheDir, rootCacheDirName + File.separator + cacheName);
+        if (!cacheBucket.exists() && !cacheBucket.mkdirs()) {
+            Log.debug(ServiceConstants.LOG_TAG, TAG, "Cannot create cache bucket.");
+            return null;
+        }
+
+        return cacheBucket;
+    }
+
+    /**
+     * Retrieves a cache file for {@code key} from {@code cacheName} directory.
+     *
+     * @param cacheName the cache sub-folder within {@code cacheRoot} where
+     *                  cache file stored for key should be fetched from
+     * @param key       the key for which the cache file should be retrieved
+     * @return the cache file for {@code key} if present; null otherwise.
+     */
+    File getCacheFile(final String cacheName, final String key) {
+        final String entryLocation = getCacheLocation(cacheName, key);
+        if (entryLocation == null) return null;
+
+        final File entryHandle = new File(entryLocation);
+        if (entryHandle.exists()) {
+            return entryHandle;
+        } else {
+            return null;
+        }
+    }
+
+    /**
+     * Retrieves a cache metadata file for {@code key} from {@code cacheName} directory.
+     *
+     * @param cacheName the cache sub-folder within {@code cacheRoot} where the metadata for
+     *                  cache file stored for key should be fetched from
+     * @param key       the key for which the cache file metadata should be retrieved
+     * @return the cache metadata for cache file stored for {@code key} if present; null otherwise.
+     */
+    Map<String, String> getCacheMetadata(final String cacheName, final String key) {
+        if (StringUtils.isNullOrEmpty(cacheName) || StringUtils.isNullOrEmpty(key)) {
+            return null;
+        }
+
+        final String metaDataLocation = getMetadataLocation(cacheName, key);
+        if (metaDataLocation == null)  {
+            Log.trace(ServiceConstants.LOG_TAG, TAG, "Metadata location for" +
```

Comment:
Can you change the log level to debug for all failures paths? We can keep it as trace in other places.